src/Components/SearchBar.js

```diff
@@ -0,0 +1,28 @@
+import React, { useState } from 'react';
+import foods from '../foods.json';
+
+function SearchBar(props) {
+
+    const [search, setSearch] = useState("");
+
+    const handleSearch = (e) => {
+        e.preventDefault();
+
+        props.setFood(() => {
+            // don't use prevFood, always the whole list [...foods] to filter
+            // avoid just 'e' for the element, "e" have full scope in this function(handleSearch(e)) (e is the event!)
+            return [...foods].filter(element => element.name.includes(e.target.value));              
```

Comment:
`return [...foods].filter(element => element.name.includes(e.target.value)); ` That way you can't search for the foods added. If you used the `foodsArr` you would have a problem as you lose foods from the list after one searching (just to keep in mind)